fix: BROS-418: UX improvements and fixes

nick-skriabin opened the pull request:
This PR addresses the following issues:

- Inability to select another region when one region is already selected AND the stage is zoomed in (coordinate system mismatch)
- Inability to continue drawing a fresh region after undo/ctrl-z
- One of the recent changes broke the multi-point transformations (move points, rotate, scale). This PR addresses this issue
- Inability to exit drawing by clicking on the last added point. This only worked for a fresh region, but not for existing ones.

https://github.com/user-attachments/assets/c49e71a2-d8e4-4dfb-a49b-5c308ff68ee9

<!--

This description MUST be filled out for a PR to receive a review. Its primary purposes are:

 - to enable your reviewer to review your code easily, and
 - to convince your reviewer that your code works as intended.

Some pointers to think about when filling out your PR description:
 - Reason for change: Description of problem and solution
 - Screenshots: All visible changes should include screenshots.
 - Rollout strategy: How will this code be rolled out? Feature flags / env var / other
 - Testing: Description of how this is being verified
 - Risks: Are there any known risks associated with this change, eg to security or performance?
 - Reviewer notes: Any info to help reviewers approve the PR
 - General notes: Any info to help onlookers understand the code, or callouts to significant portions.

You may use AI tools such as Copilot Actions to assist with writing your PR description (see https://docs.github.com/en/copilot/using-github-copilot/using-github-copilot-for-pull-requests/creating-a-pull-request-summary-with-github-copilot); however, an AI summary isn't enough by itself. You'll need to provide your reviewer with strong evidence that your code works as intended, which requires actually running the code and showing that it works.

-->

nick-skriabin committed:
Fix hovering over a shape

nick-skriabin committed:
Support selecting shapes when zoomed in

nick-skriabin committed:
Support proper undo with drawing continuation

nick-skriabin committed:
BBox, transform

nick-skriabin committed:
Finish drawing (WIP)

nick-skriabin committed:
Properly exit drawing when clicking on the last point

nick-skriabin committed:
Continue drawing + exit by clicking on last added point

nick-skriabin committed:
Cleanup

nick-skriabin committed:
Points activation fix

hlomzik commented on web/libs/editor/src/tags/control/Vector.js:
```diff
@@ -86,7 +87,14 @@ const Model = types
+
+            const area = tool?.commitDrawingRegion();
+            area && self.annotation.selectArea(area);
+          }
```
add comments describing what's going on here please. why do we search for tool? especially after undo? why do we commit region after undo? what undo is this? just not finished region as I know, right?

nick-skriabin replied:
not necessary anymore, i found simpler solution

hlomzik commented on web/libs/editor/src/components/ImageView/ImageView.jsx:
```diff
@@ -605,7 +605,7 @@ export default observer(
       });
 
-      if (hoveredRegion) {
+      if (hoveredRegion && !evt.defaultPrevented) {
```
where and how it can be prevented?

nick-skriabin replied:
anywhere in the call chain by using `evt.preventDefault()`

in this case it's prevented in the VectorRegion component in KonvaVector's onClick handler

niklub commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17823935304)

niklub committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17823935304

niklub committed:
Merge branch 'develop' into 'fb-bros-418/vector-fixes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17823935304

nick-skriabin committed:
Simplify undo logic for vector

hlomzik reviewed (approved):
Approving this to unblock fixes, but feels like interactions can be improved:
- `pointSelectionHandled` can be dropped in favor of correct event propagation stop
- methods can be made static and shorter to reduce performance hit

nick-skriabin committed:
Properly select neighbouring regions

nick-skriabin committed:
Fix region unselect

nick-skriabin committed:
Fix two-click drawing

nick-skriabin committed:
Draw points close to other points

nick-skriabin committed:
Fix transformation constraints bug

nick-skriabin committed:
Transform + point deletion

nick-skriabin committed:
Proper exiting + modifiers check

nick-skriabin committed:
Fix hover

nick-skriabin committed:
Disable transformation restrictions (tmp)

nick-skriabin committed:
Allow proper transfrmations

nick-skriabin committed:
Remove proxy nodes debug

nick-skriabin committed:
Fix formatting

niklub commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/17918552766)

niklub committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17918552766

niklub committed:
Merge branch 'develop' into 'fb-bros-418/vector-fixes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/17918552766

Commit:
Revert poetry lock
